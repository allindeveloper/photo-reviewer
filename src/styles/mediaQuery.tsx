const size = {
  laptop: "1024px",
  desktop: "2560px",
};

export const device = {
  laptop: `(min-width: ${size.laptop})`,
  desktop: `(min-width: ${size.desktop})`,
};
