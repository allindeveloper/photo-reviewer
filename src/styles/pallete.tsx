export interface IPalette {
  main: string;
}
