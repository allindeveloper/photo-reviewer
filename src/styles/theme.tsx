import { DefaultTheme } from "styled-components";

const myTheme: DefaultTheme = {
  pallete: {
    common: {
      white: "#ffffff",
      offwhite: "#f0f2f7",
      brown: "#454545",
      lightGrey: "#e8ecf2",
      normalGrey: "#e3e8f0",
    },
    primary: {
      main: "#3b55e6",
    },
  },
};

export { myTheme };
