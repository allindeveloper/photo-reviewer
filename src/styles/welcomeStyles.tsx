import styled from "styled-components";

export const StyledP = styled.p<{ paddingLeft?: number }>`
  text-transform: uppercase;
  font-weight: 600;
  margin-bottom: 25px;
  margin-top: 25px;
  padding-left: ${props => props.paddingLeft}px;
  font-weight: bolder;
  color: ${props => props.theme.pallete.primary.main};
`;
export const WelcomeEmpty = styled.div`
  text-align: center;
  width: 60%;
  margin: auto;
  opacity: 0.5;
`;
export const WelcomeActions = styled.div`
  display: flex;
  justify-content: space-between;
  height: "100%";
`;

export const ApprovedImagesSection = styled.div`
  padding-left: 30px;
  padding-right: 30px;
`;

export const BottomSection = styled.div`
  padding: 10px 30px;
`;
