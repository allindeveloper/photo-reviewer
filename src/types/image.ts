export interface IImage {
    urls: {
        raw: string;
        full: string;
        regular: string;
        small: string;
        thumb: string;
    };
    id:string;
}

export interface IApprovedImage extends IImage {}
