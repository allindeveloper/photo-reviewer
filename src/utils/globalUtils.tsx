export const getBaseUrl = () => {
  return process.env.REACT_APP_UNSPLASH_URL;
};
