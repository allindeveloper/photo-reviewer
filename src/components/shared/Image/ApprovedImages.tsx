import React from "react";
import { ScrollMenu, VisibilityContext } from "react-horizontal-scrolling-menu";
import { IApprovedImage } from "../../../types/image";
import { LeftArrow, RightArrow } from "./Arrows";
import { Card } from "./Card";

type scrollVisibilityApiType = React.ContextType<typeof VisibilityContext>;

export interface IApprovedImages {
  approvedImages: IApprovedImage[];
}

const ApprovedImages = ({ approvedImages }: IApprovedImages) => {
  const onWheel = (apiObj: scrollVisibilityApiType, ev: React.WheelEvent): void => {
    const isThouchpad = Math.abs(ev.deltaX) !== 0 || Math.abs(ev.deltaY) < 15;

    if (isThouchpad) {
      ev.stopPropagation();
      return;
    }

    if (ev.deltaY < 0) {
      apiObj.scrollNext();
    } else if (ev.deltaY > 0) {
      apiObj.scrollPrev();
    }
  };
  return (
    <>
      <div>
        <div>
          {approvedImages.length === 0 && <Card title={""} itemId={""} />}
          <ScrollMenu LeftArrow={LeftArrow} RightArrow={RightArrow} onWheel={onWheel}>
            {approvedImages.length > 0 ? (
              approvedImages.map((item, index) => (
                <Card imageUrl={item.urls.regular} title={item.id} itemId={item.id} key={item.id + index} />
              ))
            ) : (
              <div></div>
            )}
          </ScrollMenu>
        </div>
      </div>
    </>
  );
};
export default ApprovedImages;
