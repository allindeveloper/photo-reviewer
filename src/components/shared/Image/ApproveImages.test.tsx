import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { myTheme } from "../../../styles/theme";
import { IApprovedImage } from "../../../types/image";
import { setupIntersectionObserverMock } from "../../../utils/testUtils";
import ApprovedImages from "./ApprovedImages";

beforeEach(() => {
  setupIntersectionObserverMock();
});

it("Approved Images renders correctly when there are no items", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={myTheme}>
        <ApprovedImages approvedImages={[]} />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Approved Images correctly when there is a single approved Image", () => {
  const approvedImages: IApprovedImage[] = [
    {
      id: "JmFldiNGY18",
      urls: {
        raw:
          "https://images.unsplash.com/photo-1644585587551-85377778d375?ixid=MnwzMDA4MzZ8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NDQ3NTUzMjM\u0026ixlib=rb-1.2.1",
        full:
          "https://images.unsplash.com/photo-1644585587551-85377778d375?crop=entropy\u0026cs=srgb\u0026fm=jpg\u0026ixid=MnwzMDA4MzZ8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NDQ3NTUzMjM\u0026ixlib=rb-1.2.1\u0026q=85",
        regular:
          "https://images.unsplash.com/photo-1644585587551-85377778d375?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMDA4MzZ8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NDQ3NTUzMjM\u0026ixlib=rb-1.2.1\u0026q=80\u0026w=1080",
        small:
          "https://images.unsplash.com/photo-1644585587551-85377778d375?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMDA4MzZ8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NDQ3NTUzMjM\u0026ixlib=rb-1.2.1\u0026q=80\u0026w=400",
        thumb:
          "https://images.unsplash.com/photo-1644585587551-85377778d375?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMDA4MzZ8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NDQ3NTUzMjM\u0026ixlib=rb-1.2.1\u0026q=80\u0026w=200",
      },
    },
  ];
  const tree = renderer
    .create(
      <ThemeProvider theme={myTheme}>
        <ApprovedImages approvedImages={approvedImages} />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there are multiple items", () => {
  const approvedImages: IApprovedImage[] = [
    {
      id: "JmFldiNGY18",
      urls: {
        raw:
          "https://images.unsplash.com/photo-1644585587551-85377778d375?ixid=MnwzMDA4MzZ8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NDQ3NTUzMjM\u0026ixlib=rb-1.2.1",
        full:
          "https://images.unsplash.com/photo-1644585587551-85377778d375?crop=entropy\u0026cs=srgb\u0026fm=jpg\u0026ixid=MnwzMDA4MzZ8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NDQ3NTUzMjM\u0026ixlib=rb-1.2.1\u0026q=85",
        regular:
          "https://images.unsplash.com/photo-1644585587551-85377778d375?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMDA4MzZ8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NDQ3NTUzMjM\u0026ixlib=rb-1.2.1\u0026q=80\u0026w=1080",
        small:
          "https://images.unsplash.com/photo-1644585587551-85377778d375?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMDA4MzZ8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NDQ3NTUzMjM\u0026ixlib=rb-1.2.1\u0026q=80\u0026w=400",
        thumb:
          "https://images.unsplash.com/photo-1644585587551-85377778d375?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMDA4MzZ8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NDQ3NTUzMjM\u0026ixlib=rb-1.2.1\u0026q=80\u0026w=200",
      },
    },
    {
      id: "sNS-uuP01hE",
      urls: {
        raw:
          "https://images.unsplash.com/photo-1644520755723-868f336428c6?ixid=MnwzMDA4MzZ8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NDQ3NTU1MzQ\u0026ixlib=rb-1.2.1",
        full:
          "https://images.unsplash.com/photo-1644520755723-868f336428c6?crop=entropy\u0026cs=srgb\u0026fm=jpg\u0026ixid=MnwzMDA4MzZ8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NDQ3NTU1MzQ\u0026ixlib=rb-1.2.1\u0026q=85",
        regular:
          "https://images.unsplash.com/photo-1644520755723-868f336428c6?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMDA4MzZ8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NDQ3NTU1MzQ\u0026ixlib=rb-1.2.1\u0026q=80\u0026w=1080",
        small:
          "https://images.unsplash.com/photo-1644520755723-868f336428c6?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMDA4MzZ8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NDQ3NTU1MzQ\u0026ixlib=rb-1.2.1\u0026q=80\u0026w=400",
        thumb:
          "https://images.unsplash.com/photo-1644520755723-868f336428c6?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMDA4MzZ8MHwxfHJhbmRvbXx8fHx8fHx8fDE2NDQ3NTU1MzQ\u0026ixlib=rb-1.2.1\u0026q=80\u0026w=200",
      },
    },
  ];
  const tree = renderer
    .create(
      <ThemeProvider theme={myTheme}>
        <ApprovedImages approvedImages={approvedImages} />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
