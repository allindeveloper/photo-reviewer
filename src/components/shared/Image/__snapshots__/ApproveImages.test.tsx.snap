// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Approved Images correctly when there is a single approved Image 1`] = `
<div>
  <div>
    <div
      className="react-horizontal-scrolling-menu--wrapper "
      onWheel={[Function]}
    >
      <div
        className="react-horizontal-scrolling-menu--scroll-container "
        onScroll={[Function]}
      >
        <div
          className="react-horizontal-scrolling-menu--item "
          data-index={0}
          data-key="JmFldiNGY18"
        >
          <div
            className="sc-bdvvtL Xapl card"
            data-testid="approvedImages"
            role="button"
            style={
              Object {
                "margin": "0px 8px 0px 8px",
                "userSelect": "none",
                "width": "110px",
              }
            }
            tabIndex={0}
            title="JmFldiNGY18"
          >
            <div
              className="sc-gsDKAQ iqFBTl"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-check fa-lg "
                color="#ffffff"
                data-icon="check"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M438.6 105.4C451.1 117.9 451.1 138.1 438.6 150.6L182.6 406.6C170.1 419.1 149.9 419.1 137.4 406.6L9.372 278.6C-3.124 266.1-3.124 245.9 9.372 233.4C21.87 220.9 42.13 220.9 54.63 233.4L159.1 338.7L393.4 105.4C405.9 92.88 426.1 92.88 438.6 105.4H438.6z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-right fa-2x "
        color="#3b55e6"
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={
          Object {
            "alignSelf": "center",
            "cursor": "pointer",
            "display": "flex",
            "justifyContent": "center",
            "marginLeft": "5px",
            "right": "1%",
            "userSelect": "none",
          }
        }
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M64 448c-8.188 0-16.38-3.125-22.62-9.375c-12.5-12.5-12.5-32.75 0-45.25L178.8 256L41.38 118.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l160 160c12.5 12.5 12.5 32.75 0 45.25l-160 160C80.38 444.9 72.19 448 64 448z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Approved Images renders correctly when there are no items 1`] = `
<div>
  <div>
    <div
      className="sc-bdvvtL jsLcHi card"
      data-testid="approvedImages"
      role="button"
      style={
        Object {
          "margin": "0px 8px 0px 8px",
          "userSelect": "none",
          "width": "110px",
        }
      }
      tabIndex={0}
      title=""
    >
      
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-2x "
        color="#e3e8f0"
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 256c0 17.69-14.33 32.01-32 32.01H256v144c0 17.69-14.33 31.99-32 31.99s-32-14.3-32-31.99v-144H48c-17.67 0-32-14.32-32-32.01s14.33-31.99 32-31.99H192v-144c0-17.69 14.33-32.01 32-32.01s32 14.32 32 32.01v144h144C417.7 224 432 238.3 432 256z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`renders correctly when there are multiple items 1`] = `
<div>
  <div>
    <div
      className="react-horizontal-scrolling-menu--wrapper "
      onWheel={[Function]}
    >
      <div
        className="react-horizontal-scrolling-menu--scroll-container "
        onScroll={[Function]}
      >
        <div
          className="react-horizontal-scrolling-menu--item "
          data-index={0}
          data-key="JmFldiNGY18"
        >
          <div
            className="sc-bdvvtL Xapl card"
            data-testid="approvedImages"
            role="button"
            style={
              Object {
                "margin": "0px 8px 0px 8px",
                "userSelect": "none",
                "width": "110px",
              }
            }
            tabIndex={0}
            title="JmFldiNGY18"
          >
            <div
              className="sc-gsDKAQ iqFBTl"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-check fa-lg "
                color="#ffffff"
                data-icon="check"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M438.6 105.4C451.1 117.9 451.1 138.1 438.6 150.6L182.6 406.6C170.1 419.1 149.9 419.1 137.4 406.6L9.372 278.6C-3.124 266.1-3.124 245.9 9.372 233.4C21.87 220.9 42.13 220.9 54.63 233.4L159.1 338.7L393.4 105.4C405.9 92.88 426.1 92.88 438.6 105.4H438.6z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="react-horizontal-scrolling-menu--separator "
          data-index={0.1}
          data-key="JmFldiNGY18-separator"
        />
        <div
          className="react-horizontal-scrolling-menu--item "
          data-index={1}
          data-key="sNS-uuP01hE"
        >
          <div
            className="sc-bdvvtL jqqObx card"
            data-testid="approvedImages"
            role="button"
            style={
              Object {
                "margin": "0px 8px 0px 8px",
                "userSelect": "none",
                "width": "110px",
              }
            }
            tabIndex={0}
            title="sNS-uuP01hE"
          >
            <div
              className="sc-gsDKAQ iqFBTl"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-check fa-lg "
                color="#ffffff"
                data-icon="check"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M438.6 105.4C451.1 117.9 451.1 138.1 438.6 150.6L182.6 406.6C170.1 419.1 149.9 419.1 137.4 406.6L9.372 278.6C-3.124 266.1-3.124 245.9 9.372 233.4C21.87 220.9 42.13 220.9 54.63 233.4L159.1 338.7L393.4 105.4C405.9 92.88 426.1 92.88 438.6 105.4H438.6z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-right fa-2x "
        color="#3b55e6"
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={
          Object {
            "alignSelf": "center",
            "cursor": "pointer",
            "display": "flex",
            "justifyContent": "center",
            "marginLeft": "5px",
            "right": "1%",
            "userSelect": "none",
          }
        }
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M64 448c-8.188 0-16.38-3.125-22.62-9.375c-12.5-12.5-12.5-32.75 0-45.25L178.8 256L41.38 118.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l160 160c12.5 12.5 12.5 32.75 0 45.25l-160 160C80.38 444.9 72.19 448 64 448z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
